src/test/rpc/Simulate_test.cpp

```diff
@@ -0,0 +1,974 @@
+            {
+                Json::Value signer;
+                signer[jss::Account] = becky.human();
+                signer[jss::SigningPubKey] = strHex(dylan.pk().slice());
+                Json::Value signerOuter;
+                signerOuter[sfSigner] = signer;
+                tx[sfSigners].append(signerOuter);
+            }
+
+            // test with autofill
+            testTx(env, tx, validateOutput, false);
+
+            tx[sfSigningPubKey] = "";
+            tx[sfTxnSignature] = "";
+            tx[sfSequence] = env.seq(alice);
+            // transaction requires a non-base fee
+            tx[sfFee] =
+                (env.current()->fees().base * 2).jsonClipped().asString();
+            tx[sfSigners][0u][sfSigner][jss::TxnSignature] = "";
+
+            // test without autofill
+            testTx(env, tx, validateOutput);
+
+            // TODO: check that the ledger wasn't affected
+        }
+    }
+
+    void
+    testDeleteExpiredCredentials()
+    {
+        testcase("Credentials aren't actually deleted on `tecEXPIRED`");
+
+        // scenario setup
+
+        using namespace jtx;
+        Env env(*this);
+
+        Account const subject{"subject"};
+        Account const issuer{"issuer"};
+
+        env.fund(XRP(10000), subject, issuer);
+        env.close();
+
+        auto const credType = "123ABC";
+
+        auto jv = credentials::create(subject, issuer, credType);
+        uint32_t const t =
+            env.current()->info().parentCloseTime.time_since_epoch().count();
+        jv[sfExpiration.jsonName] = t;
+        env(jv);
+        env.close();
+
+        {
+            auto validateOutput = [&](Json::Value const& resp,
+                                      Json::Value const& tx) {
+                auto result = resp[jss::result];
+                checkBasicReturnValidity(
+                    result, tx, 4, env.current()->fees().base);
+
+                BEAST_EXPECT(result[jss::engine_result] == "tecEXPIRED");
+                BEAST_EXPECT(result[jss::engine_result_code] == 148);
+                BEAST_EXPECT(
+                    result[jss::engine_result_message] ==
+                    "Expiration time is passed.");
+
+                if (BEAST_EXPECT(
+                        result.isMember(jss::meta) ||
+                        result.isMember(jss::meta_blob)))
+                {
+                    Json::Value metadata;
+                    if (result.isMember(jss::meta_blob))
+                    {
+                        auto unHexed =
+                            strUnHex(result[jss::meta_blob].asString());
+                        SerialIter sitTrans(makeSlice(*unHexed));
+                        metadata = STObject(std::ref(sitTrans), sfGeneric)
+                                       .getJson(JsonOptions::none);
+                    }
+                    else
+                    {
+                        metadata = result[jss::meta];
+                    }
```

Comment:
Please add new line after bracket

Reply:
fixed in 0bbf589f53a13069448d709faa5cea88b6ee547f